### ISSUE TITLE
Packer Azure Plugin (2.1.0) fails immediately with "...context used must have a deadline attached...".

### ISSUE THREAD

**Issue by acollard** [hashicorp/packer-plugin-azure#401]
#### Overview of the Issue
Immediately after running Packer on a developer machine or build agent (Azure DevOps MS Hosted), we get the following error:
> Build 'azure-arm.web-image' errored after 254 milliseconds 3 microseconds: loading results: the context used must have a deadline attached for polling purposes, but got no deadline

This issue started 2024.04.09 as a result of 2.1.0 being released.

#### Reproduction Steps
1. Copy the build file referenced below.
2. Create a variable file (ex. variables.auto.pkvars.hcl) with required vars from build file.
3. Install Packer Azure plugin (`packer plugins install github.com/hashicorp/azure`)
4. Run Packer. `packer build -var-file=variables.auto.pkvars.hcl .`

### Plugin and Packer version
Packer v1.10.2
packer-plugin-azure_v2.1.0_x5.0_linux_amd64

### Simplified Packer Buildfile
https://gist.github.com/acollard/a4c127b16eb25551cebf7214fa26759b

### Operating system and Environment details
OS: Ubuntu 22.04.5
Architecture: X64
Included Software: https://github.com/actions/runner-images/blob/ubuntu22/20240407.1/images/ubuntu/Ubuntu2204-Readme.md

### Log Fragments and crash.log files
```
azure-arm.web-image: output will be in this color.

==> azure-arm.web-image: Running builder ...
    azure-arm.web-image: Creating Azure Resource Manager (ARM) client ...
    azure-arm.web-image: ARM Client successfully created
Build 'azure-arm.web-image' errored after 304 milliseconds 324 microseconds: loading results: the context used must have a deadline attached for polling purposes, but got no deadline

==> Wait completed after 304 milliseconds 324 microseconds

==> Some builds didn't complete successfully and had errors:
--> azure-arm.web-image: loading results: the context used must have a deadline attached for polling purposes, but got no deadline

==> Builds finished but no artifacts were created.
```

### Workaround
Rolling back to Azure plugin v2.0.5 works as expected.

**Comment by suckatrash** [comment]
I am also seeing this issue in a build pipeline with no recent changes

**Comment by JenGoldstrich** [comment]
Thanks for reporting this issue, the clear build file provided made it super easy for me to identify which call was missing the polling context, I am adding that now and will ship a patch release very shortly

**Comment by JenGoldstrich** [comment]
I have merged the fix for this issue in #402, please re-open this issue if you experience issues building with `custom_manage_image_name` sourced builds, if you experience this error with different kinds of builds, please open a new issue with the full template.

For some context in our new (non deprecated!) SDK, a polling context is required for Azure API requests, one of these was missed when querying the image name when using a `custom_managed_image` as the source.

This fix will be released in v2.1.1 shortly, thanks again for reporting this!
